src/System.Text.Json/src/System/Text/Json/ThrowHelper.cs

```diff
@@ -0,0 +1,164 @@
+                        message = string.Format(SR.DepthMustBePositive, json.CurrentDepth);
+                    else
+                        message = string.Format(SR.ArrayEndWithinObject);
+                    break;
+                case ExceptionResource.EndOfStringNotFound:
+                    message = SR.EndOfStringNotFound;
+                    break;
+                case ExceptionResource.ExpectedDigitNotFound:
+                    message = string.Format(SR.ExpectedDigitNotFound, character);
+                    break;
+                case ExceptionResource.ExpectedDigitNotFoundEndOfData:
+                    message = string.Format(SR.ExpectedDigitNotFoundEndOfData, character);
+                    break;
+                case ExceptionResource.ExpectedEndAfterSingleJson:
+                    message = string.Format(SR.ExpectedEndAfterSingleJson, character);
+                    break;
+                case ExceptionResource.ExpectedEndOfDigitNotFound:
+                    message = string.Format(SR.ExpectedEndOfDigitNotFound, character);
+                    break;
+                case ExceptionResource.ExpectedNextDigitComponentNotFound:
+                    message = string.Format(SR.ExpectedNextDigitComponentNotFound, character);
+                    break;
+                case ExceptionResource.ExpectedNextDigitEValueNotFound:
+                    message = string.Format(SR.ExpectedNextDigitEValueNotFound, character);
+                    break;
+                case ExceptionResource.ExpectedSeparaterAfterPropertyNameNotFound:
+                    message = string.Format(SR.ExpectedSeparaterAfterPropertyNameNotFound, character);
+                    break;
+                case ExceptionResource.ExpectedStartOfPropertyNotFound:
+                    message = string.Format(SR.ExpectedStartOfPropertyNotFound, character);
+                    break;
+                case ExceptionResource.ExpectedStartOfPropertyOrValueNotFound:
+                    message = SR.ExpectedStartOfPropertyOrValueNotFound;
+                    break;
+                case ExceptionResource.ExpectedStartOfValueNotFound:
+                    message = string.Format(SR.ExpectedStartOfValueNotFound, character);
+                    break;
+                case ExceptionResource.ExpectedValueAfterPropertyNameNotFound:
+                    message = SR.ExpectedValueAfterPropertyNameNotFound;
+                    break;
+                case ExceptionResource.FoundInvalidCharacter:
+                    message = string.Format(SR.FoundInvalidCharacter, character);
+                    break;
+                case ExceptionResource.InvalidEndOfJson:
+                    message = string.Format(SR.InvalidEndOfJson, json.TokenType);
+                    break;
+                case ExceptionResource.ObjectDepthTooLarge:
+                    message = string.Format(SR.ObjectDepthTooLarge, json.CurrentDepth, json.CurrentState.MaxDepth);
```

Comment:
When formatting something from resources, it's best to use `SR.Format`. See `SR.cs` but essentially it's because in AOT, where the SR.XXXXX string may be empty, it will still do something useful (append the parameters literally)

eg 
`message = SR.Format(SR.ObjectDepthTooLarge, json.CurrentDepth, json.CurrentState.MaxDepth);`
obviously this applies many other places

Reply:
Yep, @stephentoub already provided that input :)